lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +62,116 @@ def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self.content = self._extract_config_content()
+        self._validate_config_content()
+        self._seed_urls = self.content.seed_urls
+        self._num_articles = self.content.total_articles
+        self._headers = self.content.headers
+        self._encoding = self.content.encoding
+        self._timeout = self.content.timeout
+        self._should_verify_certificate = self.content.should_verify_certificate
+        self._headless_mode = self.content.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            content = json.load(f)
+        return ConfigDTO(**content)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            content = json.load(f)
+
+        seed_urls = content['seed_urls']
+        headers = content['headers']
+        total_articles_to_find_and_parse = content['total_articles_to_find_and_parse']
+        encoding = content['encoding']
+        timeout = content['timeout']
+        should_verify_certificate = content['should_verify_certificate']
+        headless_mode = content['headless_mode']
+
+        if not isinstance(seed_urls, list):
+            raise IncorrectSeedURLError
+        for url in seed_urls:
+            if not isinstance(url, str) or not re.match(r'https?://.*/', url):
+                raise IncorrectSeedURLError
+        if not isinstance(headers, dict):
+            raise IncorrectHeadersError
+        if (not isinstance(total_articles_to_find_and_parse, int)
+                or isinstance(total_articles_to_find_and_parse, bool)
+                or total_articles_to_find_and_parse < 1):
+            raise IncorrectNumberOfArticlesError
+        if total_articles_to_find_and_parse > NUM_ARTICLES_UPPER_LIMIT:
+            raise NumberOfArticlesOutOfRangeError
+        if not isinstance(encoding, str):
+            raise IncorrectEncodingError
+        if (not isinstance(timeout, int)
+                or timeout < TIMEOUT_LOWER_LIMIT
+                or timeout > TIMEOUT_UPPER_LIMIT):
+            raise IncorrectTimeoutError
+        if not isinstance(should_verify_certificate, bool) or not isinstance(headless_mode, bool):
+            raise IncorrectVerifyError
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return self.content.seed_urls
```

Comment:
this and other getting method should return values from corresponding class fields e.g. here it is self._seed_urls

Reply:
now you do not use class fields in any methods